File: VB/MVVMSchedulerApplication/ViewModel/MainViewModel.vb
#Region "#usings"
Imports System.ComponentModel

#End Region  ' #usings
Namespace MVVMSchedulerApplication.ViewModel

#Region "#mybindinglist"
    Partial Class MainViewModel

        Public Property Doctors As BindingList(Of Doctor)

        Public Property Appointments As BindingList(Of HospitalAppointment)

        Public Sub New()
            Doctors = New BindingList(Of Doctor)()
            Appointments = New BindingList(Of HospitalAppointment)()
            FillEmployees()
            FillTasks()
        End Sub

        Public Class Doctor

            Public Property Id As Object

            Public Property Name As String
        End Class

        Public Class HospitalAppointment

            Public Property PatientName As String

            Public Property Location As String

            Public Property StartTime As Date

            Public Property EndTime As Date

            Public Property InsuranceNumber As String

            Public Property FirstVisit As Boolean

            Public Property DoctorId As Object

            Public Property Notes As String
        End Class
    End Class
#End Region  ' #mybindinglist
End Namespace

File: CS/MVVMSchedulerApplication/ViewModel/MainViewModel.cs
#region #usings
using System;
using System.Collections.Generic;
using System.ComponentModel;
#endregion #usings

namespace MVVMSchedulerApplication.ViewModel
{
    #region #mybindinglist
    partial class MainViewModel
    {
        public BindingList<Doctor> Doctors { get; set; }
        public BindingList<HospitalAppointment> Appointments { get; set; }
        public MainViewModel() 
        {
            Doctors = new BindingList<Doctor>();
            Appointments = new BindingList<HospitalAppointment>();
            FillEmployees();
            FillTasks();
        }

        public class Doctor
        {
            public object Id { get; set; }
            public string Name { get; set; }
        }

        public class HospitalAppointment
        {
            public string PatientName { get; set; }
            public string Location { get; set; }
            public DateTime StartTime { get; set; }
            public DateTime EndTime { get; set; }
            public string InsuranceNumber { get; set; }
            public bool FirstVisit { get; set; }
            public object DoctorId { get; set; }
            public string Notes { get; set; }
        }
    }
    #endregion #mybindinglist
}
